Xplat/src/main/java/vazkii/botania/data/LooniumStructureLootProvider.java

```diff
@@ -0,0 +1,255 @@
+		tables.put(getLootId(BuiltinStructures.RUINED_PORTAL_NETHER), buildDelegateLootTable(BuiltInLootTables.RUINED_PORTAL));
+		tables.put(getLootId(BuiltinStructures.RUINED_PORTAL_OCEAN), buildDelegateLootTable(BuiltInLootTables.RUINED_PORTAL));
+		tables.put(getLootId(BuiltinStructures.RUINED_PORTAL_STANDARD), buildDelegateLootTable(BuiltInLootTables.RUINED_PORTAL));
+		tables.put(getLootId(BuiltinStructures.RUINED_PORTAL_SWAMP), buildDelegateLootTable(BuiltInLootTables.RUINED_PORTAL));
+		tables.put(getLootId(BuiltinStructures.SHIPWRECK), buildShipwreckLootTable());
+		tables.put(getLootId(BuiltinStructures.SHIPWRECK_BEACHED), buildShipwreckLootTable());
+		tables.put(getLootId(BuiltinStructures.STRONGHOLD),
+				// Strongholds generate up to 4 corridor chests, up to 6 crossings, and up to 2 libraries with 1 or 2 chests
+				LootTable.lootTable().withPool(LootPool.lootPool()
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.STRONGHOLD_CORRIDOR).setWeight(4))
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.STRONGHOLD_CROSSING).setWeight(6))
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.STRONGHOLD_LIBRARY).setWeight(3))
+				)
+		);
+		// skipping swamp hut, because it doesn't contain unique loot (could merge witch/cat tables, I guess)
+		tables.put(getLootId(BuiltinStructures.TRAIL_RUINS),
+				// Trail ruins have 2 common suspicious gravel for the tower top and each road section,
+				// and 6 common plus 3 rare suspicious gravel per building and for the tower bottom.
+				LootTable.lootTable().withPool(LootPool.lootPool()
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.TRAIL_RUINS_ARCHAEOLOGY_COMMON).setWeight(9))
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.TRAIL_RUINS_ARCHAEOLOGY_RARE))
+				)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_PLAINS),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_PLAINS_HOUSE, PLAINS_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_DESERT),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_DESERT_HOUSE, DESERT_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_SAVANNA),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_SAVANNA_HOUSE, SAVANNA_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_SNOWY),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_SNOWY_HOUSE, SNOWY_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.VILLAGE_TAIGA),
+				buildVillageLootTable(BuiltInLootTables.VILLAGE_TAIGA_HOUSE, TAIGA_VILLAGE_LOOT)
+		);
+		tables.put(getLootId(BuiltinStructures.WOODLAND_MANSION),
+				LootTable.lootTable().withPool(LootPool.lootPool()
+						.add(LootTableReference.lootTableReference(BuiltInLootTables.WOODLAND_MANSION).setWeight(99))
+						.add(LootItem.lootTableItem(Items.TOTEM_OF_UNDYING).setWeight(1))
+				)
+		);
+
+		var output = new ArrayList<CompletableFuture<?>>(tables.size());
```

Comment:
I think it's cleanest if this is it's own function. Or maybe this stays in the `run` function, and a separate function is made that handles creating and populating `tables`. Again, i can handle this